test/github_ex/repositories_test.exs

```diff
@@ -37,5 +37,76 @@ defmodule GithubEx.RepositoriesTest do
       assert {:error, [404, error]} = Repositories.fetch("non-existant", "repository")
       assert error["message"] == "Not Found"
     end
+
+    test "with an incorrect repository name" do
+      assert {:error, :invalid_name} = Repositories.fetch("invalidname")
+    end
+  end
+
+  describe "list/0" do
+    test "returns a list of repositories" do
+      Tesla.Mock.mock(fn
+        %{method: :get} ->
+          %Tesla.Env{
+            status: 200,
```

Comment:
o que achas de validar a url também?

Reply:
Done in [0983f0b](https://github.com/Cendrao/github_ex/pull/1/commits/0983f0bf9a29b88ce20182050ff17380428cc415).